apis/core/lib/domain/csv-source/upload.ts

```diff
@@ -37,6 +40,22 @@ export async function uploadFile({
         .addOut(schema.contentUrl, $rdf.namedNode(upload.Location))
     })
 
+  try {
+    const fileStream = await loadFile(key) as Readable
+    const head = await loadFileHeadString(fileStream)
+    const { dialect, header } = await sniffParse(head)
+
+    csvSource
+      .addOut(csvw.dialect, csvDialect => {
+        csvDialect.addOut(csvw.quoteChar, dialect.quote)
+          .addOut(csvw.delimiter, dialect.delimiter)
+          .addOut(csvw.header, true)
+          .addOut(csvw.headerRowCount, header.length)
+      })
+  } catch (err) {
+    error(err)
+  }
+
```

Comment:
What do you think, should we extract this to an other file? After the discussion of today, I think we will use this logic soon from an other place as well?

Reply:
We will extract this as soon as we need IMO :)